src/main/kotlin/se/uulm/snowballr/backend/service/CriterionService.kt

```diff
@@ -24,13 +48,137 @@ interface ICriterionService {
  *
  * @constructor Initializes the [CriterionService] with a criterion repository.
  * @param repo The repository responsible for handling persistence operations related to criteria.
+ * @param userRepo The repository responsible for handling persistence operations related to users.
+ * @param projectRepo The repository responsible for handling persistence operations related to projects.
+ * @param projectMemberRepo The repository responsible for handling persistence operations related to project members.
  */
 class CriterionService(
     private val repo: ICriterionTableRepo,
+    private val userRepo: IUserTableRepo,
+    private val projectRepo: IProjectTableRepo,
+    private val projectMemberRepo: IProjectMemberTableRepo,
 ) : ICriterionService {
+    /**
+     * Validates specific preconditions for the given criterion. A valid criterion must have
+     * either a non-null `projectId` or a non-null `createdBy`, but not both.
+     * If the precondition is violated, a [SnowballRException.FailedPreconditionException] is thrown.
+     *
+     * @param criterion The criterion object to be validated. It is expected to have a
+     *                  combination of fields that meet the required precondition.
+     * @throws SnowballRException.FailedPreconditionException If the criterion does not satisfy the required conditions.
+     */
+    private fun checkPreconditions(criterion: Criterion) {
+        val bothNull = criterion.projectId == null && criterion.createdBy == null
+        val bothNotNull = criterion.projectId != null && criterion.createdBy != null
+        if (bothNull || bothNotNull) {
+            throw SnowballRException.FailedPreconditionException("The criterion is not valid anymore.")
+        }
+    }
+
+    /**
+     * Checks if the given user has the required permissions to access or modify a project criterion.
```

Comment:
According to our `AccessType`, `access` is the general term for all CRUD operations.
```suggestion
     * Checks if the given user has the required permissions to access a project criterion.
```
Update the rest of this comment accordingly.